frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/BasePullBranchAction.java

```diff
@@ -0,0 +1,87 @@
+package com.virtuslab.gitmachete.frontend.actions.common;
+
+import static io.vavr.API.$;
+import static io.vavr.API.Case;
+import static io.vavr.API.Match;
+
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsException;
+import com.intellij.openapi.vcs.VcsNotifier;
+import git4idea.repo.GitRepository;
+import io.vavr.collection.List;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.backend.api.SyncToRemoteStatus;
+import com.virtuslab.gitmachete.frontend.datakeys.DataKeys;
+import com.virtuslab.logger.IPrefixedLambdaLogger;
+import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link DataKeys#KEY_GIT_MACHETE_REPOSITORY}</li>
+ *  <li>{@link CommonDataKeys#PROJECT}</li>
+ * </ul>
+ */
+public abstract class BasePullBranchAction extends GitMacheteRepositoryReadyAction {
+  private static final IPrefixedLambdaLogger LOG = PrefixedLambdaLoggerFactory.getLogger("frontendActions");
+
+  protected final List<SyncToRemoteStatus.Relation> PULL_ENABLING_STATUSES = List.of(
+      SyncToRemoteStatus.Relation.Behind);
+
+  /**
+   * Bear in mind that {@link AnAction#beforeActionPerformedUpdate} is called before each action.
+   * (For more details check {@link com.intellij.openapi.actionSystem.ex.ActionUtil} as well.)
```

Comment:
This comment is likely no longer needed here b/c of `GitMacheteRepositoryReadyAction`?